Automattic/sensei
# Fix blocks late initialization

We were having issues with customizing our blocks in the Course theme using style variations. After some investigation, it turned out that our blocks were initialized too late (during the `current_screen` hook for the admin and on `template_redirect` for the frontend). The [recommended](https://developer.wordpress.org/block-editor/getting-started/create-block/block-code/) way is to initialize blocks on the `init` hook.

## Proposed Changes

* Initialize the blocks on the `init` hook.
* Keep the existing logic on the admin side - initialized only on the specified post types, site editor, etc...
* On the frotnend side, previously the blocks were initialized only on the specified post types but this is no longer the case. To make style variations work, the blocks are now always initialized on the frontend (this is standard practice). The block assets are loaded on the specified post types as before.

## Testing Instructions

1. Create a course/lesson/quiz and make sure the blocks are working as before in the editor and on the frontend.
2. Make sure the blocks are working in the site editor.
3. Make sure the loaded assets in the frontend are the same as before.

## Pre-Merge Checklist
